### PR TITLE
Move the repository checks outside, so they can be

### PR DESCRIPTION
**Pull request by jsuchome**
Move the repository checks outside, so they can be reused by some other scripts.

I picked /usr/lib/suse-cloud/ as a location, that obviously needs to be handled in the spec file.

### CODE REVIEW

File: releases/development/master/extra/repo_checks [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This is a common library containing validation checks for repositories.
+
+set -e
+
```
Comment (vuntz) [inline releases/development/master/extra/repo_checks]: You need to define `die` here, don't you?

Comment (jsuchome) [inline releases/development/master/extra/repo_checks]: Oh, right. Although I was thinking about some variable way for failure (warning message + exit code), that may seem better for the upgrade script.

Comment (jsuchome) [inline releases/development/master/extra/repo_checks]: Not sure if we want to use same `die` in all cases, or let the caller decide by defining own one. This one has additional spinner handling etc.

Comment (jsuchome) [inline releases/development/master/extra/repo_checks]: I'd say: move other helper functions (`die`, ``echo_whatever`, spinner handling) to some other include, and make sure it is sourced first.

**Pull request closed without merging**